### PR TITLE
[DX-1382] Gateway release notes 5.3.2 

### PR DESCRIPTION
**Pull request by Eopayemi**
### For internal users - Please add a Jira DX PR ticket to the subject!
[DX-1382](https://tyktech.atlassian.net/browse/DX-1382)

---

### Preview Link

[preview](https://deploy-preview-4690--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/release-notes/version-5.3)

### Description

Release notes 5.3.2

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

[DX-1382]: https://tyktech.atlassian.net/browse/DX-1382?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 1]
```diff
@@ -21,6 +21,165 @@ Our minor releases are supported until our next minor comes out.
...
+Given the potential time difference between your upgrade and the release of this version, we recommend users verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.
+
+
+### Downloads
+- [Docker image to pull](https://hub.docker.com/r/tykio/tyk-gateway/tags?page=&page_size=&ordering=&name=v5.3.2)
+  - ```bash
+    docker pull tykio/tyk-gateway:v5.3.2
+    ```
+- Helm charts
+  - [tyk-charts v1.3]({{< ref "product-stack/tyk-charts/release-notes/version-1.3.md" >}})
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]: Should this be Tyk charts 1.4? @caroltyk ?

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 2]
```diff
@@ -21,6 +21,165 @@ Our minor releases are supported until our next minor comes out.
...
+
+
+### Dependencies
+
+<!--Required. Use this section to announce the following types of dependencies compatible with the release:
+Version compatibility with other components in the Tyk stack. This takes the form of a compatibility matrix and is only required for Gateway and Portal.
+3rd party dependencies and tools -->
+
+#### Compatibility Matrix For Tyk Components
+<!-- Required. Version compatibility with other components in the Tyk stack. This takes the form of a compatibility matrix and is only required for Gateway and Portal.
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md:56]: @andyo-tyk @caroltyk @slega Are there any changes to compatibility matrix required for release 5.3.2?

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 3]
```diff
@@ -21,6 +21,165 @@ Our minor releases are supported until our next minor comes out.
...
+
+#### 3rd Party Dependencies & Tools
+<!-- Required. Third-party dependencies encompass tools (GoLang, Helm etc.), databases (PostgreSQL, MongoDB etc.) and external software libraries. This section should be a table that presents the third-party dependencies and tools compatible with the release. Compatible is used in the sense of those versions tested with the releases. Such information assists customers considering upgrading to a specific release.
+
+Additionally, a disclaimer statement was added below the table, for customers to check that the third-party dependency they decide to install remains in support.
+
+An example is given below for illustrative purposes only. Tested Versions and Compatible Versions information will require discussion with relevant squads and QA. -->
+
+
+| Third Party Dependency                                       | Tested Versions        | Compatible Versions    | Comments |
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md:77]: @andyo-tyk @caroltyk Are there any changes to 3rd party dependencies for release 5.3.2?

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR LGTM. Following needs to be updated before release:

- Release Date
- Compatibility Matrix and 3rd Party Dependencies neede considering if anything needs updated for 5.3.2 release
- Left a comment/query if link to Tyk Charts 1.4 needs to be added in downloads section, instead of Tyk Charts 1.3

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 4]
```diff
@@ -21,6 +21,128 @@ Our minor releases are supported until our next minor comes out.
...
+<!-- This section should be a bullet point list that describes the issues fixed in the release. For each fixed issue explain:
+- What problem the issue caused
+- How was the issue fixed
+- Link to (new) documentation created as a result of a fix. For example, a new configuration parameter may have been introduced and documented for the fix
+- For OSS - Link to the corresponding issue if possible on GitHub to allow the users to see further info.
+Each change log item should be expandable. The first line summarises the changelog entry. It should be then possible to expand this to reveal further details about the changelog item. This is achieved using HTML as shown in the example below. -->
+<ul>
+<li>
+<details>
+<summary>Custom Domain Handling Enhancement</summary>
```
Comment (dcs3spp) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]: Remove capitalisations for summary headings

**Review by dcs3spp — COMMENTED** [review]
@elizabeth Thanks for submitting changelog. Remember to remove capitilaisation of words in changelog

**Review by dcs3spp — APPROVED** [review]

**Comment by dcs3spp** [comment]
@andyo-tyk @agata-wit Ok to release?

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 5]
```diff
@@ -21,6 +21,135 @@ Our minor releases are supported until our next minor comes out.
...
+<li>
+<details>
+<summary>Gateway service discovery issue with consul</summary>
+
+Addressed an issue in service discovery where IP:port returned by Consul wasn't parsed correctly on the Gateway side, leading to errors when proxying requests to the service. The issue primarily occurred with IP:port responses, while valid domain names were unaffected.
+</details>
+</li>
+<li>
+<details>
+<summary>Resolved UDG Nested Field Mapping Issue</summary>
```
Comment (agata-wit) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]:
```suggestion
<summary>Resolved Universal Data Graph Nested Field Mapping Issue</summary>
```

**Review by agata-wit — DISMISSED** [review]

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 6]
```diff
@@ -21,6 +21,135 @@ Our minor releases are supported until our next minor comes out.
...
+<li>
+<details>
+<summary>Custom domain handling enhancement</summary>
+
+Addressed an issue where APIs with custom domains weren't prioritised correctly, potentially causing conflicts with APIs lacking custom domains on the same listen path. APIs with custom domain names are now loaded before others, ensuring proper fallback behaviour.
+</details>
+</li>
+<li>
+<details>
+<summary>Improved error handling in event handler template tests</summary>
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]: Remove this one

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 7]
```diff
@@ -21,6 +21,135 @@ Our minor releases are supported until our next minor comes out.
...
+- How was the issue fixed
+- Link to (new) documentation created as a result of a fix. For example, a new configuration parameter may have been introduced and documented for the fix
+- For OSS - Link to the corresponding issue if possible on GitHub to allow the users to see further info.
+Each change log item should be expandable. The first line summarises the changelog entry. It should be then possible to expand this to reveal further details about the changelog item. This is achieved using HTML as shown in the example below. -->
+<ul>
+<li>
+<details>
+<summary>Custom domain handling enhancement</summary>
+
+Addressed an issue where APIs with custom domains weren't prioritised correctly, potentially causing conflicts with APIs lacking custom domains on the same listen path. APIs with custom domain names are now loaded before others, ensuring proper fallback behaviour.
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]:
```suggestion
Addressed an issue where an API with a custom domain might not be invoked if another API with the same listen path but no custom domain was also deployed on the Gateway. Now APIs with custom domain names are loaded first, so requests will be checked against these first before falling back to APIs without custom domains.
```

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 8]
```diff
@@ -21,6 +21,135 @@ Our minor releases are supported until our next minor comes out.
...
+<summary>Improved error handling in event handler template tests</summary>
+
+Enhanced error handling in webhook event template rendering to ensure proper logging and skipping of the event when errors occur, preventing invalid data from being sent.
+</details>
+</li>
+<li>
+<details>
+<summary>Gateway service discovery issue with consul</summary>
+
+Addressed an issue in service discovery where IP:port returned by Consul wasn't parsed correctly on the Gateway side, leading to errors when proxying requests to the service. The issue primarily occurred with IP:port responses, while valid domain names were unaffected.
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]:
```suggestion
Addressed an issue in service discovery where an IP:port returned by Consul wasn't parsed correctly on the Gateway side, leading to errors when proxying requests to the service. The issue primarily occurred with IP:port responses, while valid domain names were unaffected.
```

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 9]
```diff
@@ -21,6 +21,135 @@ Our minor releases are supported until our next minor comes out.
...
+<li>
+<details>
+<summary>Resolved Universal Data Graph Nested Field Mapping Issue</summary>
+
+Fixed an issue with nested field mapping in UDG when used with GraphQL (GQL) operations for a field's data source. Previously, querying only the mentioned field resulted in an error, but querying alongside another 'normal' field from the same level worked without issue.
+</details>
+</li>
+<li>
+<details>
+<summary>Added control for context variables in OAS API definition</summary>
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]:
```suggestion
<summary>Added control over access to context variables from middleware when using Tyk OAS APIs</summary>
```

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 10]
```diff
@@ -21,6 +21,135 @@ Our minor releases are supported until our next minor comes out.
...
+
+Introduced control over context variables in OAS API Definitions, allowing users to enable or disable access to context variables in middleware and plugins.
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]:
```suggestion
Addressed a potential issue when working with Tyk OAS APIs where request context variables are automatically made available to relevant Tyk and custom middleware. We have introduced a control in the Tyk OAS API definition to disable this access if required.
```

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md [hunk 11]
```diff
@@ -21,6 +21,135 @@ Our minor releases are supported until our next minor comes out.
...
+<!-- This section should be a bullet point list that describes the issues fixed in the release. For each fixed issue explain:
+- What problem the issue caused
+- How was the issue fixed
+- Link to (new) documentation created as a result of a fix. For example, a new configuration parameter may have been introduced and documented for the fix
+- For OSS - Link to the corresponding issue if possible on GitHub to allow the users to see further info.
+Each change log item should be expandable. The first line summarises the changelog entry. It should be then possible to expand this to reveal further details about the changelog item. This is achieved using HTML as shown in the example below. -->
+<ul>
+<li>
+<details>
+<summary>Custom domain handling enhancement</summary>
```
Comment (andyo-tyk) [inline tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]:
```suggestion
<summary>APIs with common listen paths but different custom domains</summary>
```

**Review by andyo-tyk — CHANGES_REQUESTED** [review]

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md (hunk 3 above)
Comment (andyo-tyk) [reply tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md:77]: Not from a Core perspective - this is a bug fix release.

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md (hunk 2 above)
Comment (andyo-tyk) [reply tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md:56]: N/A for Core

**Review by andyo-tyk — APPROVED** [review]
Thanks for making the changes. 👍

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md (hunk 2 above)
Comment (carolwkc) [reply tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md:56]: none from platform

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md (hunk 3 above)
Comment (carolwkc) [reply tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md:77]: No changes

File: tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md (hunk 1 above)
Comment (carolwkc) [reply tyk-docs/content/product-stack/tyk-gateway/release-notes/version-5.3.md]: updated

**Review by carolwkc — APPROVED** [review]

**Pull request merged**